### PR TITLE
add proxy config factory

### PR DESCRIPTION
**Pull request by pvormste**
This PR does following changes:
 - adds a proxy config factory
 - moves federation config factory into `graphql` package
 - refactors federation config factory to make it more flexible
 - shares logic between proxy and federation config factory

### CODE REVIEW

File: examples/federation/gateway/gateway.go [hunk 1]
```diff
@@ -75,7 +74,7 @@ func (g *Gateway) Ready() {
 // Error handling is not finished.
```
Comment (jensneuse) [inline examples/federation/gateway/gateway.go:74]: This comment is supicious

File: pkg/graphql/config_factory_federation.go [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+type FederationEngineConfigFactory struct {
+	httpClient        *http.Client
+	dataSourceConfigs []graphqlDataSource.Configuration
+	schema            *Schema
+}
+
+func (f *FederationEngineConfigFactory) SetMergedSchemaFromString(mergedSchema string) (err error) {
+	f.schema, err = NewSchemaFromString(mergedSchema)
+	if err != nil {
+		return fmt.Errorf("set merged schema: %s", err.Error())
```
Comment (jensneuse) [inline pkg/graphql/config_factory_federation.go:56]: should we annotate errors with info on the origin? e.g. a reference to the FederationEngineConfigFactory

File: pkg/graphql/config_factory_proxy.go [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+	}
+}
+
+type ProxyUpstreamConfig struct {
+	URL           string
+	Method        string
+	StaticHeaders http.Header
+}
+
+type ProxyEngineConfigFactory struct {
```
Comment (jensneuse) [inline pkg/graphql/config_factory_proxy.go:29]: description missing

File: pkg/graphql/config_factory_federation.go [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+		optFunc(&options)
+	}
+
+	return &FederationEngineConfigFactory{
+		httpClient:        options.httpClient,
+		dataSourceConfigs: dataSourceConfigs,
+	}
+}
+
+type FederationEngineConfigFactory struct {
```
Comment (jensneuse) [inline pkg/graphql/config_factory_federation.go:47]: description missing

File: pkg/graphql/engine_config_v2.go [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+	schema                   *Schema
+	plannerConfig            plan.Configuration
+	websocketBeforeStartHook WebsocketBeforeStartHook
+}
+
+func NewEngineV2Configuration(schema *Schema) EngineV2Configuration {
+	return EngineV2Configuration{
+		schema: schema,
+		plannerConfig: plan.Configuration{
+			DefaultFlushInterval: 0,
```
Comment (jensneuse) [inline pkg/graphql/engine_config_v2.go:21]: We should have a global with a reasonable default flush interval.
E.g. 1 second.

**Review by jensneuse — CHANGES_REQUESTED** [review]

**Review by spetrunin — APPROVED** [review]

**Review by jensneuse — APPROVED** [review]

**Pull request merged**